src/traces/pie/helpers.js

```diff
@@ -25,3 +25,16 @@ exports.formatPieValue = function formatPieValue(v, separators) {
     }
     return Lib.numSeparate(vRounded, separators);
 };
+
+exports.getFirstFilled = function getFirstFilled(array, indices) {
```

Comment:
For `arrayOk` attributes, as well as slice color, I chose to take the first non-empty item corresponding to each label. I suppose in principle we could refine that further to the first *valid* item for each label?

Reply:
Picking the first non-empty item is fine :+1: